### ISSUE TITLE
Router Packets not decodable 

### ISSUE THREAD

**Issue by xeb** [exo-explore/exo#75]
I am receiving the following error on multiple nodes in my home network.

```
Task exception was never retrieved
future: <Task finished name='Task-9003' coro=<GRPCDiscovery.on_listen_message() done, defined at /projects/exo/exo/networking/grpc/grpc_discovery.py:104> exception=UnicodeDecodeError('utf-8', b'\xbf\xa2\x04\x00\x00\x01\x00\x06,\xc8\x1b\x13\x98\xe0\x00\x05\x00\x08MikroTik\x00\x07\x00\x0f6.49.1 (stable)\x00\x08\x00\x08MikroTik\x00\n\x00\x04\xd7|\xf3\x00\x00\x0b\x00\tX4N0-HRA8\x00\x0c\x00\x08RB760iGS\x00\x0e\x00\x01\x00\x00\x10\x00\rbridge/ether5\x00\x11\x00\x04\xc0\xa8X\x01', 0, 1, 'invalid start byte')>
Traceback (most recent call last):
  File "/projects/exo/exo/networking/grpc/grpc_discovery.py", line 105, in on_listen_message
    message = json.loads(data.decode('utf-8'))
                         ^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xbf in position 0: invalid start byte
```

Unfortunately, it's not a matter of just ignoring the errors (e.g. `data.decode('utf-8', errors='ignore')`) since the result is not able to be decoded with JSON.

Seems like the root cause is my MikroTik router. Here are some details about it:
```
[admin@MikroTik] > system resource print
                   uptime: 26w2d16h41m15s
                  version: 6.49.1 (stable)
               build-time: Nov/17/2021 10:06:00
         factory-software: 6.46.4
              free-memory: 211.5MiB
             total-memory: 256.0MiB
                      cpu: MIPS 1004Kc V2.15
                cpu-count: 4
            cpu-frequency: 880MHz
                 cpu-load: 0%
           free-hdd-space: 3860.0KiB
          total-hdd-space: 16.0MiB
  write-sect-since-reboot: 5665524
         write-sect-total: 6025588
               bad-blocks: 0%
        architecture-name: mmips
               board-name: hEX S
                 platform: MikroTik
```

Here is a GIST that can decode the result of the discovery call: https://gist.github.com/xeb/f871811220293e54bb26ad51d85523d0

So I **think** that exo's `main.py` is basically just getting packets from my router & showing the error. I suspect the right behavior is to ignore it & update [on_listen_message](https://github.com/exo-explore/exo/blob/main/exo/networking/grpc/grpc_discovery.py#L104) to be more resilient to traffic that isn't from another exo node, right?

Messages that are not JSON, but are binary for example.

Happy to submit a PR! Just curious about the preferred intended behavior.

**Comment by AlexCheema** [comment]
Thanks so much for the detailed issue!

The way you've laid this out and your proposed solution sounds great. We should indeed ignore messages that didn't come from another exo node, including binary messages. Feel free to submit a PR with the changes you mentione.

**Comment by AlexCheema** [comment]
@xeb

**Comment by xeb** [comment]
ACK'd! Will try to get a PR shortly. I wasn't thrilled with a try...except...

**Comment by xeb** [comment]
Fixed in PR #90
